src/system/SystemPacketBuffer.cpp

```diff
@@ -253,11 +264,23 @@ uint16_t PacketBuffer::ReservedSize() const
 
 uint8_t * PacketBuffer::ReserveStart()
 {
+#if CHIP_SYSTEM_CONFIG_USE_LWIP
+    if (!(PBUF_STRUCT_DATA_CONTIGUOUS(this)))
+    {
+        return reinterpret_cast<uint8_t *>(this->payload);
```

Comment:
Any reason not to just return `Start()` here?

Reply:
Changed